### ISSUE TITLE
Incorrectly printed suggestion for `map(<f>).unwrap_or(false)` on multiple lines

### ISSUE THREAD

**Issue by oscargus** [rust-lang/rust-clippy#13242]
### Summary

The following clippy message has an incorrect suggestion:

```
warning: called `map(<f>).unwrap_or(false)` on an `Option` value
    --> surfer\src/main.rs:1994:9
     |
1994 | /         self.waves
1995 | |             .as_ref()
1996 | |             .map(|w| w.inner.is_fully_loaded())
1997 | |             .unwrap_or(false)
     | |_____________________________^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or
help: use `is_some_and(<f>)` instead
     |
1996 -             .map(|w| w.inner.is_fully_loaded())
1996 +             .is_some_and(|w| w.inner.is_fully_loaded())
```

as the `.unwrap_or(false)` should also be removed.

However, running `--fix`, it works correctly so it is just the printing that is wrong.

### Reproducer

``` rust
fn main() {
    println!("Hello, world! {}", foo());
}

fn foo() -> bool {
    let really_long_label_really_long_label = Some(4);
    let another_really_long_label = really_long_label_really_long_label
        .as_ref()
        .map(|w| *w > 3)
        .unwrap_or(false);
    another_really_long_label
}
```

(It is crucial that map and unwrap_or ends up on different lines...)

This should be checked with `-W clippy::pedantic`.

### Version

```text
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
binary: rustc
commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23
commit-date: 2024-08-06
host: x86_64-pc-windows-msvc
release: 1.80.1
LLVM version: 18.1.7
```

**Comment by Jarcho** [comment]
This no longer occurs. Don't know which version of rustc actually fixed the display issue.
